### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by romppane**
Centralized error handling added to the project combined with logging of requests and responces! Ended up leaving everything logging related to the server.ts file as the streams kept leaking to the console? :D Perhaps exporting open streams just isn't a very good idea?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,11 +24,18 @@
    },
    "homepage": "https://github.com/romppane/Bulletin_Board#readme",
    "dependencies": {
+      "@hapi/boom": "^7.4.2",
+      "@types/boom": "^7.2.1",
       "@types/express": "^4.17.0",
+      "@types/hapi__boom": "^7.4.0",
+      "@types/moment-timezone": "^0.5.12",
+      "@types/morgan": "^1.7.35",
```
Comment (henrikra) [inline package.json]: All type packages should be in `devDependencies`. When installing types you can use `-D` like this for example `npm i -D @types/boom`

File: package.json [hunk 2]
```diff
@@ -24,11 +24,18 @@
    },
    "homepage": "https://github.com/romppane/Bulletin_Board#readme",
    "dependencies": {
+      "@hapi/boom": "^7.4.2",
+      "@types/boom": "^7.2.1",
       "@types/express": "^4.17.0",
+      "@types/hapi__boom": "^7.4.0",
```
Comment (henrikra) [inline package.json]: Is types for boom installed twice? I am referring to `@types/hapi__boom` and `@types/boom`

File: server/helpers/validation.ts [hunk 3]
```diff
@@ -45,10 +45,8 @@ export const validatePost = (req: Request, res: Response, next: NextFunction) =>
     // See if the newly made Post is valid
     validate(newpost).then(errors => { // errors is an array of validation errors
         if (errors.length > 0) {
-            console.log("validation failed. errors: ", errors);
-            res.status(400).send(errors);
+            next(new Boom("Validation error", {statusCode : 400}));
```
Comment (henrikra) [inline server/helpers/validation.ts]: Rather use the Boom's helpers function. For example here `next(Boom.badRequest('Validation error');)`. Same thing in all places where you create Booms

File: package.json (hunk 2 above)
Comment (romppane) [reply package.json]: Looks like I forgot to uninstall @types/boom which I installed simultaniously with boom, which to my big surprice wasn't the real dependency needed? Leaving it uninstalled this time as I swap the dependencies to the dev mode though!

File: server/helpers/validation.ts (hunk 3 above)
Comment (romppane) [reply server/helpers/validation.ts]: That definitely adds readability! Implementing the changes now!

**Comment by henrikra** [comment]
Looks good! Approved 👍

**Pull request merged**